test/functional/deprecate_warning_tests.js

```diff
@@ -0,0 +1,116 @@
+'use strict';
+const exec = require('child_process').exec;
+const chai = require('chai');
+const expect = chai.expect;
+const sinon = require('sinon');
+const sinonChai = require('sinon-chai');
+require('mocha-sinon');
+chai.use(sinonChai);
+
+const utils = require('../tools/utils');
+const ClassWithLogger = utils.ClassWithLogger;
+const ClassWithoutLogger = utils.ClassWithoutLogger;
+const ClassWithUndefinedLogger = utils.ClassWithUndefinedLogger;
+const ensureCalledWith = utils.ensureCalledWith;
+
+describe('Deprecation Warnings', function() {
+  beforeEach(function() {
+    this.sinon.stub(console, 'error');
+  });
+
+  const defaultMessage = ' is deprecated and will be removed in a later version.';
+
+  it('node --no-deprecation flag should suppress all deprecation warnings', {
+    metadata: { requires: { node: '>=6.0.0' } },
+    test: function(done) {
+      exec(
+        'node --no-deprecation ./test/tools/deprecate_warning_test_program.js',
+        (err, stdout, stderr) => {
+          expect(err).to.be.null;
+          expect(stdout).to.be.empty;
+          expect(stderr).to.be.empty;
+          done();
+        }
+      );
+    }
+  });
+
+  it('node --trace-deprecation flag should print stack trace to stderr', {
+    metadata: { requires: { node: '>=6.0.0' } },
+    test: function(done) {
+      exec(
+        'node --trace-deprecation ./test/tools/deprecate_warning_test_program.js',
+        (err, stdout, stderr) => {
+          expect(err).to.be.null;
+          expect(stdout).to.be.empty;
+          expect(stderr).to.not.be.empty;
+
+          const split = stderr.split('\n');
```

Comment:
I'm not clear what this parsing is doing. What is happening here?

Reply:
This parsing was just to clean up the warning message (it contained some information like path that wouldn't be consistent across different machines) and separate out the subsequent stack trace into individual lines. I wanted to check that:
- stderr started with the correct warning message (lines 55-57)
- the rest of it was a valid stack trace, i.e. each line beginning with the word 'at' (lines 61-63)

Reply:
Ok, add a comment that explains what this is doing.